src/main/java/com/faforever/client/map/MapService.java

```diff
@@ -631,6 +637,28 @@ public void destroy() {
 
   }
 
+
+  /**
+   * TODO: Make this function as general (universal) for all images from any place (examples: mod image preview)
+   */
+  public void showMapInCloseUp(String mapName) {
+    Rectangle2D screenBounds = Screen.getPrimary().getBounds();
+    double imageOptimalSize = screenBounds.getHeight() * 0.7;
+    double centerScreenX = screenBounds.getWidth() / 2 - imageOptimalSize / 2;
+    double centerScreenY = screenBounds.getHeight() / 2 - imageOptimalSize / 2;
+
+    ImageView mapImageView = new ImageView(loadPreview(mapName, PreviewSize.LARGE));
+    mapImageView.setFitHeight(imageOptimalSize);
+    mapImageView.setSmooth(true);
+    mapImageView.setPreserveRatio(true);
+
+    Popup popup = new Popup();
+    popup.setAutoHide(true);
+    popup.setAutoFix(true);
+    popup.getScene().setRoot(new StackPane(mapImageView));
+    popup.show(StageHolder.getStage(), centerScreenX, centerScreenY);
+  }
```

Comment:
I agree that this should be moved to some other util service and made general. The Services should not be creating UI elements.

Ideally the mapService would load the image and then a util class would be called that creates the popup all from the Controller Class that has the click function.